lnwallet/wallet.go

```diff
@@ -972,10 +973,26 @@ func (l *LightningWallet) handleFundingCounterPartySigs(msg *addCounterPartySigs
 			)
 			if err != nil {
 			}
+
+			ctx, cancel := context.WithCancel(context.Background())
```

Comment:
since our goal is simply to break on shutdown, we should only need to make one `context.Context` for the whole `LightningWallet`. the `ctx` and `cancel` can be tracked as member variables, and `cancel` only needs to invoked when `Stop`is called. similar comment applies for router

Reply:
We could, but that seems like a big deviation from our normal pattern? I like sticking with the simple `cancel chan` tbh.

Reply:
what's the normal pattern? having one context for all concurrent wallet/router `GetUtxo` requests seems more in line with how we would otherwise pass the object's `quit` channel, like in #2716?

Reply:
I think having a `context.Background` makes sense for the struct itself. We can then use specific deadlines/cancels by creating a context specific for that call, like we do with the integration tests.

Reply:
> I like sticking with the simple cancel chan tbh.

Do you mean `quit` chans or `cancel()` funcs?

> I think having a context.Background makes sense for the struct itself. We can then use specific deadlines/cancels by creating a context specific for that call, like we do with the integration tests.

Agreed, though it's not necessary to create a cancellable context for each call when our only need is to cancel them all simultaneously. If in the future we want to be able to cancel/timeout individual calls, then creating a subcontext would be appropriate. In the meantime it is extra goroutine overhead

atm i'm kind of in favor of #2716 for simplicity, it matches the existing patterns w/in the codebase, and accomplishes the same task w/ less overhead

Reply:
> Do you mean quit chans or cancel() funcs?

I meant as in #2716 

> what's the normal pattern?

That is " it matches the existing patterns w/in the codebase", and I'm also in favor of #271 for now.